### PR TITLE
[RE-930]: all the methods that accepted field id can accept field name

### PR DESCRIPTION
**Pull request by konorlevich**
and make hardcoded custom field id redundant and deprecated

### CODE REVIEW

File: src/Issue.php [hunk 1]
```diff
@@ -377,12 +380,45 @@ public function getFieldValue(string $field_id, array $expand = [])
...
+    /**
+     * Returns fields id for any known field in the issue
+     *
+     * @param $fieldName string name of field, e.g. 'Created', 'Summary', 'Build name' and so on
+     *
+     * @return string empty string for unknown field
+     */
+    public function getFieldId(string $fieldName): string
+    {
+        $names = $this->BaseIssue->names??[];
```
Comment (rosmanov) [inline src/Issue.php]:
```suggestion
        $names = $this->BaseIssue->names ?? [];
```

File: src/Issue.php [hunk 2]
```diff
@@ -377,12 +380,45 @@ public function getFieldValue(string $field_id, array $expand = [])
...
     }
 
+
+    /**
+     * Returns fields id for any known field in the issue
+     *
+     * @param $fieldName string name of field, e.g. 'Created', 'Summary', 'Build name' and so on
+     *
+     * @return string empty string for unknown field
+     */
```
Comment (rosmanov) [inline src/Issue.php]:
```suggestion
    /**
     * Retrieves the ID of a field by its name if the field exists in the issue.
     *
     * @param string $fieldName The name of the field, e.g., 'Created,' 'Summary,' 'Build name,' and so on.
     *
     * @return string The field ID, or an empty string if the field name is unknown.
     */
```

File: src/Issue.php [hunk 3]
```diff
@@ -377,12 +380,45 @@ public function getFieldValue(string $field_id, array $expand = [])
...
+        foreach ($names as $fieldId => $name) {
+            if ($name == $fieldName) {
+                return $fieldId;
+            }
+        }
+        return "";
+    }
+
+    /**
+     * @deprecated use {@link getEditableFieldIdByName getEditableFieldIdByName} instead
```
Comment (rosmanov) [inline src/Issue.php:406]: We don't expect the user to call this method, do we? If it's deprecated from the beginning, then we probably shouldn't add it?

File: src/Issue.php [hunk 4]
```diff
@@ -377,12 +380,45 @@ public function getFieldValue(string $field_id, array $expand = [])
...
+     * @param array $editMeta
+     * @param string $fieldName
```
Comment (rosmanov) [inline src/Issue.php]:
```suggestion
```

File: src/Issue.php [hunk 5]
```diff
@@ -419,29 +455,39 @@ public function getFieldValueByName(string $fieldName, array $expand = [])
      * This is useful e.g. for text fields with formatters (e.g. wiki-), to get the same layout as in issue view page
      * without tricks and hacks.
      *
-     * @param string $field_id - ID of field to get (e.g. 'customfield_12345' or 'summary')
+     * @param string $fieldName ID of name of field to get (e.g. 'customfield_12345' or 'Summary')
```
Comment (rosmanov) [inline src/Issue.php]:
```suggestion
     * @param string $fieldName Field ID or name (e.g., 'customfield_12345' or 'Summary')
```

File: src/Issue.php [hunk 6]
```diff
@@ -419,29 +455,39 @@ public function getFieldValueByName(string $fieldName, array $expand = [])
...
+        return $this->getBaseIssue([\Badoo\Jira\REST\Section\Issue::EXP_RENDERED_FIELDS])
+            ->renderedFields
+            ->{$fieldId} ?? null;
     }
 
     /**
      * Get date/time field value as UNIX timestamp. strtotime() is used to parse the value of the field.
      *
-     * @param string $field_id - ID of field to parse, e.g. 'created', 'updated' and so on
+     * @param string $fieldName ID of name of field to parse, e.g. 'Created', 'updated', "customfield_12345" and so on
```
Comment (rosmanov) [inline src/Issue.php]:
```suggestion
     * @param string $fieldName ID or name of the field to parse (e.g. 'Created', 'updated', "customfield_12345" and so on).
```

File: src/REST/Section/Issue.php [hunk 7]
```diff
@@ -78,9 +78,9 @@ public function get(string $issue_key, array $fields = [], array $expand = [], a
         if (!empty($fields)) {
             $args['fields'] = implode(',', $fields);
         }
-        if (!empty($expand)) {
-            $args['expand'] = implode(',', $expand);
-        }
+
+        $args['expand'] = implode(',', array_unique(array_merge($expand + ["names"])));
```
Comment (rosmanov) [inline src/REST/Section/Issue.php]: We need use either `+` or `array_merge()`, but not both.

File: src/REST/Section/Issue.php [hunk 8]
```diff
@@ -124,9 +124,8 @@ public function search(
         if (!empty($fields)) {
             $args['fields'] = $fields;
         }
-        if (!empty($expand)) {
-            $args['expand'] = $expand;
-        }
+
+        $args['expand'] = array_unique(array_merge($expand + ["names"]));
```
Comment (rosmanov) [inline src/REST/Section/Issue.php]: We need either `+` or `array_merge`, but not both.

**Review by rosmanov — CHANGES_REQUESTED** [review]

File: src/REST/Section/Issue.php (hunk 7 above)
Comment (konorlevich) [inline src/REST/Section/Issue.php]:
```suggestion
        $args['expand'] = implode(',', array_unique(array_merge($expand, ["names"])));
```

File: src/REST/Section/Issue.php (hunk 8 above)
Comment (konorlevich) [inline src/REST/Section/Issue.php]:
```suggestion
        $args['expand'] = array_unique(array_merge($expand, ["names"]));
```

File: src/Issue.php (hunk 3 above)
Comment (konorlevich) [reply src/Issue.php:406]: I added this method a few days ago

Now I can't just remove it ¯\_(ツ)_/¯

**Pull request closed without merging**